agent/taskresource/interface.go

```diff
@@ -0,0 +1,45 @@
+// Copyright 2014-2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package taskresource
+
+import (
+	"encoding/json"
+	"time"
+)
+
+// ResourceStatus is an enumeration of valid states of task resource lifecycle
+type ResourceStatus int32
+
+// TaskResource is a wrapper for task level resource methods we need
+type TaskResource interface {
+	// SetDesiredStatus sets the desired status of the resource
+	SetDesiredStatus(ResourceStatus)
+	// GetDesiredStatus gets the desired status of the task
```

Comment:
nit: `// GetDesiredStatus gets the desired status of the resource`
and few other comments too